CHECLabPy/core/spectrum_fitter.py

```diff
@@ -200,10 +200,15 @@ def load_config(self, path):
                 if 'initial' in d:
                     ini = c
                     self.initial[ini] = d['initial'].pop(c, self.initial[ini])
+                    if(self.initial[ini] is not None):
+                        self.initial[ini] = float(self.initial[ini])
                 if 'limits' in d:
                     lim = "limit_" + c
                     list_ = d['limits'].pop(c, self.limits[lim])
-                    self.limits[lim] = tuple(list_)
+                    if(isinstance(list_,list)):
+                        self.limits[lim] = tuple([float(l) for l in list_])
+                    else:
+                        self.limits[lim] = list_
```

Comment:
I realized that my first attempt did not work if the parameter did not have a default value and it was not set in the config file. This less elegant solution checks whether the result from the pop is valid for conversion.